### DISCUSSION TITLE
Question about the Script in the rgb-schemata

### DISCUSSION THREAD

**Discussion by cymqqqq** [RGB-WG/rgb#84] [Q&A] [answered]
Hi there, I found the code in the rgb-schemata:
<img width="550" alt="image" src="https://github.com/RGB-WG/rgb/assets/8555534/835f13ea-39ec-44dd-a829-31ea72beed41">
So, for the script: none!() here, what can we do for the script and how to execute it? To Write a program and then compile it into a script or something else?

**Comment by dr-orlovsky** [comment, marked as answer]
Here is an example of how a script can be written today, taken from a schema implementing RGB21 NFT interface for a single token:

https://github.com/RGB-WG/rgb-schemata/blob/d0565eed1baca5bc7afbaf9f6bf7da2c6cb40416/src/uda.rs#L46-L155

Today, we use RGB assembler for that (which can be compiled using rust macro `rgbasm!`), however, a higher-level language called "Contractum" is in development and in 2024 we plan to have some first releases of it.
